### PR TITLE
Add a test for the user index.

### PR DESCRIPTION
**Pull request by DFurnes**
#### Changes

This pull request adds a functional test for the `getAllUsers` function in the Northstar API client, which gets us some super high-level functional testing on a lot of the pieces that go into that: OAuth token requests/entities/storage, API response model and collection classes.

It also adds some little helper classes to help make mocked responses a bit more easily.

---

For review: @weerd

### CODE REVIEW

File: tests/NorthstarTest.php [hunk 1]
```diff
@@ -35,23 +33,57 @@ public function testNorthstarApiClient()
...
-                XIuZGV2OjgwMDAiLCJhdWQiOiJ0cnVzdGVkLXRlc3QtY2xpZW50IiwianRpIjoiZTA2YmMwOGVkYTk2NGJmZDA2NjNmNzkzODE0OGRlN
-                jNiNjM3OTFiYzE5MWE1MTBiZjgwNmVjMmEwZmI2NTNjNzUyMWE5YjNhMWYzY2M2NzIiLCJpYXQiOjE0NzYyMDEzMzEsIm5iZiI6MTQ3N
-                jIwMTMzMSwiZXhwIjoxNDc2MjA0OTMxLCJzdWIiOiIiLCJyb2xlIjoiIiwic2NvcGVzIjpbXX0.aFyfvihTvumdWPgtfXxKQZYKVGJw-
-                DnvUZO-XXNUgKYy8ngq0OTHMV00_VEqJVHPZVZ1FQv7sq-LcxpSpEM-VgfPsNUVuhJ8E3nX63ntTQQiv_CFBvippb1LJqVhfE7gzaKQc
-                eatw5dT25nDlzRQIrUY1kz1exGHUozvkHAeJ_g',
-                'expires_in' => 3600,
-            ])),
-            new Response(200, ['Content-Type' => 'application/json'], json_encode([
+            new JwtResponse(),
+            new JsonResponse([
```
Comment (weerd) [inline tests/NorthstarTest.php:37]: no need for `use` statements at top for these?

**Review by weerd — COMMENTED** [review]
One quick question.

File: tests/NorthstarTest.php (hunk 1 above)
Comment (DFurnes) [reply tests/NorthstarTest.php:37]: They're not namespaced, so nope.

Comment (weerd) [reply tests/NorthstarTest.php:37]: ah, jes. ok

**Review by weerd — APPROVED** [review]
👍

**Pull request merged**